src/mongo/delegates/Views.class.php

```diff
@@ -588,15 +589,6 @@ protected function doJoins($source, $joins, &$dest, $from, $contextAlias, $build
                     }
                     if (!(isset($ruleset['condition']) && $collection->count($ruleset['condition'])==0))
                     {
-                        if ($buildImpactIndex && !isset($dest[_IMPACT_INDEX])) $dest[_IMPACT_INDEX] = array();
```

Comment:
Oh, I see!  Sorry, I didn't realize we weren't getting back anything *new* with the cursor.